tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -754,6 +759,97 @@ static byte TestCoveredIndex1(Span<byte> data)
             return r;
         }
 
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Covered index in-loop redundant bounds check elimination")]
+        public static void CoveredIndex2(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = TestCoveredIndex2(s, 0, length);
+                }
+                return result;
+            },
+            "CoveredIndex2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestCoveredIndex2(Span<byte> data, int start, int end)
+        {
+            byte x = 0;
+            byte y = 0;
+
+            for (var idx = start; idx < end; idx++)
+            {
+                x ^= data[idx];
+
+                if (idx == 100)
```

Comment:
May want to invert this condition so that the case whose optimization we're hoping to check is the one that usually executes.